### PR TITLE
Update to the Repository layout for v1.0.0

### PR DESCRIPTION
**Pull request by samvarankashyap**
- Rebased the develop branch with repository layout proposed by @herlo 
- Pulled recent changes in develop branch , resolved conflicts
- Updated repository layout by moving the old file around 

This PR is a prerequisite for all the upcoming PR's

### CODE REVIEW

**Comment by herlo** [comment]
centos-paas-test

**Comment by herlo** [comment]
The tests here won't pass until the jenkins jobs pull from the right directories. Please ignore test results for the time being.

File: linchpin/templates/linchpin_config.yml.j2 [hunk 1]
```diff
@@ -5,7 +5,7 @@ inventoryfolder_path: "{{ inventoryfolder_path | default(playbook_dir+'/inventor
 async: {{ async | default(false) }}
 async_timeout : {{ async_timeout | default(1000) }}
 no_output : {{ no_output | default(false) }}
-schema: "{{ schema | default(playbook_dir+'/ex_schemas/schema_v2.json') }}"
+schema: "{{ schema | default(playbook_dir+'/schemas/schema_v2.json') }}"
```
Comment (herlo) [inline linchpin/templates/linchpin_config.yml.j2]: This should be schema_v3.json

File: linchpin/tests/playbooks/linchpin_config.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ inventory_playbooks:
 async: false
 async_timeout : 1000
 no_output : false
-schema: "ex_schemas/schema_v2.json"
+schema: "schemas/schema_v2.json"
```
Comment (herlo) [inline linchpin/tests/playbooks/linchpin_config.yml]: This should be schema_v3.json

File: setup.py [hunk 3]
```diff
@@ -1,31 +1,24 @@
...
-            rel_file = os.path.join(rel_dir, file_name)
-            if any(ext in rel_file for ext in ignore_dir):
-                continue
-            file_set.append(rel_file)
-    return file_set
 
 setup(
     name='linchpin',
-    version='0.8.6',
+    version='0.9.1',
```
Comment (herlo) [inline setup.py]: Let's set this to 1.0.0pre1 for now.

**Comment by samvarankashyap** [comment]
@herlo Updated according to the suggestions

**Comment by herlo** [comment]
@samvarankashyap apparently, pypi doesn't like pre1, so let's use the following naming conventions going forward.

a = alpha (eg. version='1.0.0a1')
b = beta (eg. version='1.0.0b1')
rc = release candidate (eg. version='1.0.0rc1')

I believe these all work, though I have not tested the 'beta' value yet.

**Pull request merged**